Some fixed to interfaces and processes

lozingaro opened the pull request:
I found some broken links in processes.md and fixed it.

Moreover some old fixes from the operation/interfaces documentation done month ago is ready to be merged with the new documentation.

Hope it will be useful.

Commit:
update of communicatoin ports - interfaces

Following https://github.com/jolie/docs/pull/5 - interfaces as design tool

Commit:
processes borken links fixed for communication ports and zip file of examples

Commit:
Merge branch 'master' of git://github.com/jolie/docs

fmontesi commented on documentation/basics/communication_ports/interfaces.md:
```diff
@@ -5,36 +5,68 @@ Jolie defines two types of operations:
-The type declarations of both request and response messages are explained further in the data types subsection below.
+The type declarations of both `request` and `response` messages are explained further in the [data types](data_types.md) section.
 
+## Interface as design tool
```
This doesn't look like a fix, rather an addition. And I'm not sure I understand it. The example looks very abstract at this point.

lozingaro replied:
Oops.. Well, this was an addition from @bmaschio docs actually, I agree with you that this is not a fix at all, and for the time being it should be removed. I am adding more broken link's fixes to the docs.

Commit:
removing the "interface as design tool" section

## Interface as design tool

In Jolie the concept of interface is strongly coupled to the use of communication ports. Yet it can be used as a designing tool in order to define microservices application's domain. 

For instance let's think to start modelling a mathematical service. 
The example below show a possible implementation of the `MathInterface`.

<pre><code class="language-jolie code">
interface MathInterface {
    RequestResponse:
   sum(SumRequest)(SumResponse),
     cosine(cosineRequest)(cosineResponse)
}
</code></pre>

In *Jolie* it is possible to define the same operation in several `interfaces`, as a result we can separate the arithmetic operations from the trigonometric one as shown in our new version of the above example. 

<pre><code class="language-jolie code">
interface ArithmeticInterface {
    RequestResponse: 
        sum(SumRequest)(SumResponse), 
        cosine(cosineRequest)(cosineResponse)
}

interface TrigonometricInterface {
    RequestResponse: 
        cosine(cosineRequest)(cosineResponse)
}
</code></pre>

From a practical point of view a microservice can implement both interfaces and expose them on the identical port, re-thinking the interfaces as domain delimiters.

thesave commented:
Unfortunately, I just had time now to review this assignment.
Since the pull request had become stale in the meantime, I scanned the website for broken links and fixed them all. I think we can close this.

@szingaro sorry for the inconvenience.
Please, in case you think there was information in your request that should be integrated into the docs (excluding the broken links), can you kindly synchronize with the current version of the docs and re-issue a pull-request for that?
